src/content/blog/astro-470.mdx

```diff
@@ -0,0 +1,70 @@
+---
+title: "Astro 4.7"
+description: "Astro 4.7 is out now! This release includes major improvements to the API for making toolbar apps, a new way to keep yourself up to date, and more."
+publishDate: "April 25, 2024"
+authors:
+  - erika
+  - ema
+  - matthew
+  - nate
+  - bjorn
+coverImage: "/src/content/blog/_images/astro-470/post-header-4.7.webp"
+socialImage: "/src/content/blog/_images/astro-470/og-image-4.7.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+import updateCheckerImage from "/src/content/blog/_images/astro-470/update-checker.webp";
+import BlogContentImage from "/src/components/BlogContentImage.astro"
+
+<BlogSummary>
+  Astro 4.7 is now available! This release includes major improvements to the API for making toolbar apps, more ways to keep your Astro project up to date, and more.
+</BlogSummary>
+
+Full release highlights include:
+
+- [**Major improvements to the Dev Toolbar API**](#dev-toolbar-api-improvements)
+- [**Update checker**](#update-checker)
+- [**`allowJs: true` for strictest TypeScript preset**](#allowjs-true-for-strictest-typescript-preset)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Dev Toolbar API improvements
+
+Astro 4.7 includes major improvements to the API for making toolbar apps. One of our main goal this release was to make it easier to build and maintain toolbar apps, and we're excited to share these improvements with you!
+
+- `defineToolbarApp` helper that makes it easier to define toolbar apps, this is similar to the `defineConfig` helper for defining Astro config, or `defineMiddleware` for defining middleware.
+- `app` and `server` helpers to make it easier to interact with the toolbar and the server.
+- New starter project for building toolbar apps, to help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
+- [Revamped documentation for building toolbar apps](https://docs.astro.build/en/reference/dev-toolbar-app-reference/).
+- [All new recipe guiding you through building a toolbar app from scratch](https://docs.astro.build/en/recipes/making-toolbar-apps/), including how to use JSX frameworks like React or Preact in your app.
```

Comment:
```suggestion
- a `defineToolbarApp()` helper that makes it easier to define toolbar apps. This pattern should feel familiar as it is similar to `defineConfig()` in your Astro config and `defineMiddleware()` for defining middleware.
- new `app` and `server` helpers to make it easier to send and receive messages between the toolbar and the server.
- a brand new starter project for building toolbar apps. This will help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
- a [recipe guiding you step-by-step through building your own toolbar app](https://docs.astro.build/en/recipes/making-toolbar-apps/) from scratch, including how to build your app using JSX frameworks like React or Preact.
- revamped [documentation of the Dev Toolbar App API](https://docs.astro.build/en/reference/dev-toolbar-app-reference/), with all examples updated to show using the new methods.
```